src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -222,1149 +215,1311 @@ static constexpr const int32_t window_editor_object_selection_animation_divisor[
+        if (selected_object == -1)
+            return;
 
-    intent = Intent(INTENT_ACTION_REFRESH_SCENERY);
-    context_broadcast_intent(&intent);
-}
+        ObjectListItem* listItem = &_listItems[selected_object];
+        uint8_t object_selection_flags = *listItem->flags;
+        if (object_selection_flags & OBJECT_SELECTION_FLAG_6)
+            return;
 
-/**
- *
- *  rct2: 0x006AAFAB
- */
-static void WindowEditorObjectSelectionMouseup(rct_window* w, rct_widgetindex widgetIndex)
-{
-    switch (widgetIndex)
-    {
-        case WIDX_CLOSE:
-            window_close(w);
-            if (gScreenFlags & SCREEN_FLAGS_EDITOR)
-            {
-                finish_object_selection();
-            }
-            if (gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER)
-            {
-                game_unload_scripts();
-                title_load();
-            }
-            break;
-        case WIDX_FILTER_RIDE_TAB_ALL:
-            _filter_flags |= FILTER_RIDES;
-            gConfigInterface.object_selection_filter_flags = _filter_flags;
-            config_save_default();
+        Invalidate();
 
-            FilterUpdateCounts();
-            VisibleListRefresh(w);
+        const CursorState* state = context_get_cursor_state();
+        OpenRCT2::Audio::Play(OpenRCT2::Audio::SoundId::Click1, 0, state->position.x);
 
-            w->selected_list_item = -1;
-            w->scrolls[0].v_top = 0;
-            w->Invalidate();
-            break;
-        case WIDX_FILTER_RIDE_TAB_TRANSPORT:
-        case WIDX_FILTER_RIDE_TAB_GENTLE:
-        case WIDX_FILTER_RIDE_TAB_COASTER:
-        case WIDX_FILTER_RIDE_TAB_THRILL:
-        case WIDX_FILTER_RIDE_TAB_WATER:
-        case WIDX_FILTER_RIDE_TAB_STALL:
-            _filter_flags &= ~FILTER_RIDES;
-            _filter_flags |= (1 << (widgetIndex - WIDX_FILTER_RIDE_TAB_TRANSPORT + _numSourceGameItems));
-            gConfigInterface.object_selection_filter_flags = _filter_flags;
-            config_save_default();
+        if (gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER)
+        {
+            if (!window_editor_object_selection_select_object(0, INPUT_FLAG_EDITOR_OBJECT_SELECT, listItem->repositoryItem))
+                return;
 
-            FilterUpdateCounts();
-            VisibleListRefresh(w);
+            // Close any other open windows such as options/colour schemes to prevent a crash.
+            window_close_all();
+            // window_close(w);
 
-            w->selected_list_item = -1;
-            w->scrolls[0].v_top = 0;
-            w->frame_no = 0;
-            w->Invalidate();
-            break;
+            // This function calls window_track_list_open
+            ManageTracks();
+            return;
+        }
 
-        case WIDX_ADVANCED:
-            w->list_information_type ^= 1;
-            w->Invalidate();
-            break;
+        flags = INPUT_FLAG_EDITOR_OBJECT_1 | INPUT_FLAG_EDITOR_OBJECT_SELECT_OBJECTS_IN_SCENERY_GROUP;
```

Comment:
this is your mistake this should be a new variable. This is modifying the window flags when it shouldnt be.